src/dataIntegration/metabotools/calculateLODs.m

```diff
@@ -14,14 +14,30 @@
 %   lod_mM:            Detection limits in mM
 %
 % .. Author: - Maike K. Aurich 27/05/15
+%  Modified by Loic Marx, November 2018
 
-lod_gL = lod_ngmL* 0.000001;
+% input checking
+if length(lod_ngmL) == 1
+    
+    lod_ngmL = repmat(lod_ngmL, 1, length(theo_mass)); 
+    
+    % throw a warning
+    warning('The inputs have different size but we managed to fix this problem');
```

Comment:
I'd suggest to remove the warning and update the documentation, that the inputs can either be a single value which is used for every lod or a vector of individual lods.

Reply:
Sounds good. @loicmarx, may you do that? 👍